std/process.d

```diff
@@ -1050,54 +1050,103 @@ private Pid spawnProcessPosix(scope const(char[])[] args,
 
                 static if (!__traits(compiles, closefrom))
                 {
-                    // FIXME: This implementation crashes the system when RLIMIT_NOFILE
-                    // has a big value. For a possible solution see:
-                    // https://github.com/dlang/phobos/pull/8990
-                    void fallback (int lowfd) {
-                        // NOTE: malloc() and getrlimit() are not on the POSIX async
-                        // signal safe functions list, but practically this should
-                        // not be a problem. Java VM and CPython also use malloc()
-                        // in its own implementation via opendir().
-                        import core.stdc.stdlib : malloc;
-                        import core.sys.posix.poll : pollfd, poll, POLLNVAL;
+                    void fallback (int lowfd)
+                    {
+                        import core.sys.posix.dirent : dirent, opendir, readdir, closedir, DIR;
+                        import core.sys.posix.unistd : close;
+                        import core.sys.posix.stdlib : atoi, malloc, free;
                         import core.sys.posix.sys.resource : rlimit, getrlimit, RLIMIT_NOFILE;
 
                         // Get the maximum number of file descriptors that could be open.
                         rlimit r;
                         if (getrlimit(RLIMIT_NOFILE, &r) != 0)
-                        {
                             abortOnError(forkPipeOut, InternalError.getrlimit, .errno);
-                        }
+
                         immutable maxDescriptors = cast(int) r.rlim_cur;
 
-                        immutable maxToClose = maxDescriptors - lowfd;
+                        // Missing druntime declaration
+                        pragma(mangle, "dirfd")
+                        extern(C) nothrow @nogc int dirfd(DIR* dir);
```

Comment:
Would be desirable to do this properly in druntime. Also why does this need `pragma(mangle)`?

Reply:
Because it's a nested function, and won't get properly mangled for C. Probably should just go outside the function.

Reply:
redone. We can worry about druntime later.

Reply:
eh, I'm going to put it back, because that requires me to do more imports, which kinda sucks.